scripts/build_shapes.py

```diff
@@ -698,13 +709,394 @@ def _process_func_pop(gadm_idxs):
+            window_topleft_latitude,
+        )
+
+    # Set an empty numpy array with the dimensions of the country .tif file
+    # np_map_ID will contain an ID for each location (undefined is 0)
+    # ID corresponds to a specific geometry in country_rows
+    np_map_ID = np.zeros((y_axis_len, x_axis_len))
+
+    # List to contain the mappings of id to GADM_ID
+    id_to_GADM_ID = []
+
+    # Loop the country_rows geoDataFrame
+    for i in range(len(country_rows)):
+        # Set the current geometry
+        cur_geometry = country_rows.iloc[i]["geometry"]
+
+        # In windowed mode we check if bounds of geometry overlap the window
+        if windowed:
+            latitude_min = cur_geometry.bounds[1]
+            latitude_max = cur_geometry.bounds[3]
+
+            # In the following cases we don't have to continue the loop
+            # If the geometry is above the window
+            if latitude_min > window_topleft_latitude:
+                continue
+            # If the geometry is below the window
+            if latitude_max < window_botright_latitude:
+                continue
+
+        # Generate a mask for the specific geometry
+        temp_mask = rasterio.features.geometry_mask(
+            [cur_geometry],
+            (y_axis_len, x_axis_len),
+            transform=affine_transform,
+            all_touched=True,
+            invert=True,
+        )
+
+        # Map the values of counter value to np_map_ID
+        np_map_ID[temp_mask] = i + 1
+
+        # Store the id -> GADM_ID mapping
+        id_to_GADM_ID.append([i + 1, country_rows.iloc[i]["GADM_ID"]])
+
+    # Return np_map_ID as type 'H' np.ushort
+    # 'H' -> https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.ushort
+    # This lowers memory usage, note: ID has to be within the range [0,65535]
+    return np_map_ID.astype("H"), pd.DataFrame(id_to_GADM_ID).set_index(0)
+
+
+def compute_population(country_rows, WorldPop_inputfile, out_logging=False):
+    """
+    Function computes the population for the given country rows
+    -------
+    Inputs:
+        country_rows: geoDataFrame filled with geometries and their GADM_ID
+        WorldPop_inputfile: file location of worldpop file
+        out_logging: Boolean to enable logging
+    --------
+    Outputs:
+        df_pop_count: Dataframe with columns
+            - "pop" containing population of GADM_ID region
+            - "GADM_ID"
+    """
+    # Get the features of the worldpop input file
+    transform, worldpop_dim = get_worldpop_features(WorldPop_inputfile)
+
+    worldpop_y_dim, worldpop_x_dim = worldpop_dim
+
+    # Rasterio doesn't support lower than float32 readout
+    # Hence reading the file will take up: nbytes = 4 * y_dim * x_dim
+    expected_bytes_input_read = 4 * worldpop_y_dim * worldpop_x_dim
+
+    # Introduce a max byte size to avoid overfilling RAM
+    # Ensure worldpop_byte_limit > 883 * 10**6 (minimum memory for 'US')
+    worldpop_byte_limit = 1024 * 10**6
```

Comment:
This value shall be a config parameter or an argument of the function

Reply:
Now incorporated with 418207e28d023382afe3c21460e08c9595e2d822